Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -1739,20 +1701,193 @@ bool is_non_manifold_vertex(typename boost::graph_traits<PolygonMesh>::vertex_de
+    verts.push_back(v2);
+  }
+
+  template<typename OutputIterator>
+  void dump(OutputIterator out)
+  {
+    typedef std::pair<const vertex_descriptor, std::vector<vertex_descriptor> > Pair_type;
+    BOOST_FOREACH(const Pair_type& p, collections)
+      *out++ = p.second;
+  }
+
+  void dump(Emptyset_iterator) { }
+
+  std::map<vertex_descriptor, std::vector<vertex_descriptor> > collections;
+};
+
+} // end namespace internal
+
+template <typename PolygonMesh, typename VPM, typename ConstraintMap>
+typename boost::graph_traits<PolygonMesh>::vertex_descriptor
+create_new_vertex_for_sector(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor sector_begin_h,
+                             typename boost::graph_traits<PolygonMesh>::halfedge_descriptor sector_last_h,
+                             PolygonMesh& pm,
+                             const VPM& vpm,
+                             const ConstraintMap& cmap)
+{
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor    vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor  halfedge_descriptor;
+
+  vertex_descriptor old_vd = target(sector_begin_h, pm);
+  vertex_descriptor new_vd = add_vertex(pm);
+  put(vpm, new_vd, get(vpm, old_vd));
+
+  put(cmap, new_vd, true);
+
+  set_halfedge(new_vd, sector_begin_h, pm);
+  halfedge_descriptor h = sector_begin_h;
+  do
+  {
+    set_target(h, new_vd, pm);
+
+    if(h == sector_last_h)
+      break;
+    else
+      h = prev(opposite(h, pm), pm);
+  }
+  while(h != sector_begin_h); // for safety
+  CGAL_assertion(h != sector_begin_h);
+
+  return new_vd;
+}
+
+template <typename PolygonMesh, typename NamedParameters>
+std::size_t make_umbrella_manifold(typename boost::graph_traits<PolygonMesh>::halfedge_descriptor h,
+                                   PolygonMesh& pm,
+                                   internal::Vertex_collector<PolygonMesh>& dmap,
+                                   const NamedParameters& np)
+{
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor    vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor  halfedge_descriptor;
+
+  typedef typename GetVertexPointMap<PolygonMesh, NamedParameters>::type VertexPointMap;
+  VertexPointMap vpm = choose_param(get_param(np, internal_np::vertex_point),
+                                    get_property_map(vertex_point, pm));
+
+  typedef typename boost::lookup_named_param_def<internal_np::vertex_is_constrained_t,
+                                                 NamedParameters,
+                                                 Constant_property_map<vertex_descriptor, bool> // default (no constraint pmap)
+                                                 >::type                  VerticesMap;
+  VerticesMap cmap = choose_param(get_param(np, internal_np::vertex_is_constrained),
+                                  Constant_property_map<vertex_descriptor, bool>(false));
```

Comment:
I think I already mentioned it but the bool must be a template parameter with partial specialization to enable the compiler to optimize the calls to get.

Reply:
Can I do that for all `CGAL::Constant_property_map` in another PR?

Reply:
Sure